finance_dl/radius.py

```diff
@@ -0,0 +1,274 @@
+  exist, it will be created.
+
+- `profile_dir`: Optional.  If specified, must be a `str` that specifies the
+  path to a persistent Chrome browser profile to use.  This should be a path
+  used solely for this single configuration; it should not refer to your normal
+  browser profile.  If not specified, a fresh temporary profile will be used
+  each time.  It is highly recommended to specify a `profile_dir` to avoid
+  having to manually enter a multi-factor authentication code each time.
+
+- `earliest_history_date`: Optional.  If specified, must be a `datetime.date`
+  specifying the earliest UTC date for which to retrieve data.
+
+- `max_history_days`: Optional.  If `earliest_history_date` is not specified,
+  this must be a positive `int` specifying the number of days of history to
+  retrieve, starting from the previous UTC day.  Defaults to `365*4`.  If
+  `earliest_history_date` is specified, `max_history_days` has no effect.
+
+Output format:
+==============
+
+QFX file.
+
+Example:
+========
+
+    def CONFIG_radius():
+        return dict(
+            module='finance_dl.radius',
+            credentials={
+                'username': 'XXXXXX',
+                'password': 'XXXXXX',
+                'account_uid' : 'XXXXXX-XXXX-XXXX-XXXXXXXXXXXX',
+            },
+            output_directory=os.path.join(data_dir, 'radius'),
+
+            # profile_dir is optional but highly recommended to avoid having to
+            # enter multi-factor authentication code each time.
+            profile_dir=os.path.join(profile_dir, 'chrome')
+        )
+
+
+Look up the account_uid by logging into web portal, navigating to the transactions page 
+and searching the web page source code for the pattern:
+ download-url="/accounts/XXXXXX-XXXX-XXXX-XXXXXXXXXXXX"
+
+
+Interactive shell:
+==================
+
+From the interactive shell, type: `self.run()` to start the scraper.
+
+"""
+
+import io
+import urllib.parse
+import re
+import dateutil.parser
+import datetime
+import logging
+import os, time, shutil, glob, re
+from selenium.webdriver.common.by import By
+from selenium.common.exceptions import NoSuchElementException
+from selenium.webdriver.support.ui import Select
+from selenium.webdriver.common.keys import Keys
+
+from . import scrape_lib
+
+from IPython import embed
+
+logger = logging.getLogger('radius_scrape')
+
+netloc_re = r'^([^\.@]+\.)*radiusbank.com$'
+
+
+def check_url(url):
+    result = urllib.parse.urlparse(url)
+    if result.scheme != 'https' or not re.fullmatch(netloc_re, result.netloc):
+        raise RuntimeError('Reached invalid URL: %r' % url)
+
+
+standard_date_format = '%m/%d/%Y'
+
+class Scraper(scrape_lib.Scraper):
+    def __init__(self, credentials, output_directory,
+                 earliest_history_date=None, max_history_days=30,
+                 **kwargs):
+        """
+        @param earliest_history_date: Earliest UTC date for which to retrieve
```

Comment:
Perhaps just download all available history by default?

Reply:
Simplified this by downloading entire year.